src/Symfony/Component/Messenger/Bridge/GooglePubSub/Transport/GooglePubSubTransportFactory.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\GooglePubSub\Transport;
+
+use Symfony\Component\Messenger\Transport\Serialization\SerializerInterface;
+use Symfony\Component\Messenger\Transport\TransportFactoryInterface;
+use Symfony\Component\Messenger\Transport\TransportInterface;
+
+/**
+ * @author Richard Henkenjohann <richardhenkenjohann@googlemail.com>
+ */
+class GooglePubSubTransportFactory implements TransportFactoryInterface
+{
+    public function createTransport(string $dsn, array $options, SerializerInterface $serializer): TransportInterface
+    {
+        unset($options['transport_name']);
+
+        return new GooglePubSubTransport(Connection::fromDsn($dsn, $options), $serializer);
+    }
+
+    public function supports(string $dsn, array $options): bool
+    {
+        return 0 === strpos($dsn, 'gps://');
```

Comment:
What about `return 0 === strpos($dsn, 'gps://') || 0 === strpos($dsn, 'gcp://')`?

Reply:
There is also Google Pub/Sub Lite which uses a different API and thus is not compatible. GCP (= Google Cloud Platform) would've been too generic for me.